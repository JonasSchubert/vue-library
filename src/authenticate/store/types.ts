export const ActionTypes = {
  login: 'login',
  logout: 'logout',
  validateToken: 'validate-token'
};

export const GetterTypes = {
  error: 'error',
  isAdministrator: 'isAdministrator',
  isLoading: 'isLoading',
  isUser: 'isUser',
  loggedIn: 'loggedIn',
  token: 'token',
  userRoles: 'userRoles'
};

export const ModuleType = 'authenticate';

export const MutationTypes = {
  setError: 'setError',
  setIsLoading: 'setIsLoading',
  setToken: 'setToken',
  setUserRoles: 'setUserRoles'
};

export const RouteTypes = {
  authenticate: 'authenticate'
};
