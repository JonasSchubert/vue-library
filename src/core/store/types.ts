export const GetterTypes = {
  isAnythingLoading: 'isAnythingLoading'
};

export const MutationTypes = {
  decreaseRequests: 'decreaseRequests',
  increaseRequests: 'increaseRequests'
};
