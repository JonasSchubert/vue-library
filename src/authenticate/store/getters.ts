import { GetterTree } from 'vuex';
import { GetterTypes } from './types';
import { UserRole } from '../enums';
import { AuthenticateState, LoginResponse } from '../models';

export const createGetters = <TRootState, T extends LoginResponse>(customGetters?: GetterTree<AuthenticateState<T>, TRootState>): GetterTree<AuthenticateState<T>, TRootState> => ({
  [GetterTypes.error]: (state: AuthenticateState<T>): Error | undefined => state.error,
  [GetterTypes.isAdministrator]: (state: AuthenticateState<T>): boolean => UserRole.toArray(state.data?.role ?? UserRole.Null).includes(UserRole.Administrator),
  [GetterTypes.isLoading]: (state: AuthenticateState<T>): boolean => state.isLoading,
  [GetterTypes.isUser]: (state: AuthenticateState<T>): boolean => UserRole.toArray(state.data?.role ?? UserRole.Null).includes(UserRole.User),
  [GetterTypes.loggedIn]: (state: AuthenticateState<T>): boolean => !!state.data?.token,
  [GetterTypes.token]: (state: AuthenticateState<T>): string | undefined => state.data?.token,
  [GetterTypes.userRoles]: (state: AuthenticateState<T>): UserRole[] => UserRole.toArray(state.data?.role ?? UserRole.Null),
  ...customGetters
});
