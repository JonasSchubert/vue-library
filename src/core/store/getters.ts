import { GetterTree } from 'vuex';
import { BaseState } from '../models';
import { GetterTypes } from './types';

export const createGetters = <T extends BaseState>(): GetterTree<T, T> => ({
  [GetterTypes.isAnythingLoading]: (state: T): boolean => state.requests > 0
});
