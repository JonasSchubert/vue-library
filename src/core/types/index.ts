export * from './ga-colors.type';
