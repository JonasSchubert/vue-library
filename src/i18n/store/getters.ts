import { LocaleMessages } from 'vue-i18n';
import { GetterTree } from 'vuex';
import { I18nState } from '../models';
import { GetterTypes } from './types';

export const createGetters = <T>(): GetterTree<I18nState, T> => ({
  [GetterTypes.applicationToken]: (state: I18nState): string | undefined => state.applicationToken,
  [GetterTypes.availableLocales]: (state: I18nState): string[] => state.availableLocales,
  [GetterTypes.currentLocale]: (state: I18nState): string => state.currentLocale,
  [GetterTypes.error]: (state: I18nState): Error | undefined => state.error,
  [GetterTypes.locales]: (state: I18nState): LocaleMessages => state.locales
});
