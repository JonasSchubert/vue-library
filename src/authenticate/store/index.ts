import { Module } from 'vuex';
import { AuthenticateModuleSetup, AuthenticateState, LoginResponse } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createAuthenticateState } from './state';

export const createAuthenticateModule = <TRootState, T extends LoginResponse>({ cookieKeyAuthenticationToken, customGetters, daysTilExpiredAuthenticationCookie }: AuthenticateModuleSetup<TRootState, T>): Module<AuthenticateState<T>, TRootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(customGetters),
  mutations: createMutations({ cookieKeyAuthenticationToken, daysTilExpiredAuthenticationCookie }),
  state: createAuthenticateState()
});
