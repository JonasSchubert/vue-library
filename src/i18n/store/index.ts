import { Module } from 'vuex';
import { I18nModuleConfig, I18nState } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

export const createModule = <T>(config: I18nModuleConfig): Module<I18nState, T> => ({
  namespaced: true,
  actions: createActions<T>(config),
  getters: createGetters<T>(),
  mutations: createMutations(config),
  state: createState(config)
});
