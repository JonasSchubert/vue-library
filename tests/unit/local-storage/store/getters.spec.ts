import { GetterTree } from 'vuex';
import {
  LocalStorageHistory, LocalStorageSettings, LocalStorageState
} from '@/local-storage/models';
import { createGetters } from '@/local-storage/store/getters';
import { GetterTypes } from '@/local-storage/store/types';

describe('getters', () => {
  const state: LocalStorageState = {
    history: {
      lastRoutes: []
    },
    settings: { isDarkTheme: true }
  };

  const getters: GetterTree<LocalStorageState, LocalStorageState> = createGetters<LocalStorageState>();

  describe(`${GetterTypes.history}`, () => {
    test('should return history as expected', () => {
      // Arrange && Act
      const actual: LocalStorageHistory = getters[GetterTypes.history](state, getters, state, {});

      // Assert
      expect(actual).toMatchSnapshot();
    });
  });

  describe(`${GetterTypes.isDarkTheme}`, () => {
    test('should return isDarkTheme as expected', () => {
      // Arrange && Act
      const actual: boolean = getters[GetterTypes.isDarkTheme](state, getters, state, {});

      // Assert
      expect(actual).toBeTruthy();
    });
  });

  describe(`${GetterTypes.lastRoutes}`, () => {
    test('should return lastRoutes as expected', () => {
      // Arrange && Act
      const actual: any[] = getters[GetterTypes.lastRoutes](state, getters, state, {});

      // Assert
      expect(actual).toMatchSnapshot();
    });
  });

  describe(`${GetterTypes.settings}`, () => {
    test('should return settings as expected', () => {
      // Arrange && Act
      const actual: LocalStorageSettings = getters[GetterTypes.settings](state, getters, state, {});

      // Assert
      expect(actual).toMatchSnapshot();
    });
  });
});
