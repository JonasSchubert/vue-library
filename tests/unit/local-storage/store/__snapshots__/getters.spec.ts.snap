// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getters history should return history as expected 1`] = `
Object {
  "lastRoutes": Array [],
}
`;

exports[`getters lastRoutes should return lastRoutes as expected 1`] = `Array []`;

exports[`getters settings should return settings as expected 1`] = `
Object {
  "isDarkTheme": true,
}
`;
